citest/google_session/google_session.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import requests
+
+from bs4 import BeautifulSoup
+
+class GoogleSession:
+  """Utility class for automating Google authentication.
+
+  Automates both the Google account login and scope approval to create an
```

Comment:
If this isnt about talking to OAuth, but general use of an athenticated session, then how does it relate to citest.service_testing.HttpAgent? Should the HttpAgent support these types of sessions? If so I'd suggest refactoring that. We can talk about it if you want.